### PR TITLE
Web3.js: added maxRetries option to SendOptions

### PR DESCRIPTION
**Pull request by RTISCOOL**
This pull request implements a solution for issue #21071 

#### Problem (by @brianlong) 
There are certain use-cases, like NFT drops, where the default RPC retry behavior is counterproductive. NFT drops are designed to sell out quickly. After that point, 100% of transactions will fail and we do not want to retry TXs.

Developers should have the option to designate the max number of retries when submitting a TX. The sendTransaction RPC API method includes a maxRetries parameter, but that parameter is not present in Web3.js SendOptions.

#### Summary of Changes
- Added maxRetries parameter to SendOptions
- Added logic to sendEncodedTransaction that adds maxRetries to the rpc request config if available. 
- Added maxRetries parameter to ConfirmOptions for use in sendAndConfirmTransaction and sendAndConfirmRawTransaction

```javascript
  // Sign transaction, broadcast, and confirm
  const signature = await web3.sendAndConfirmTransaction(
    connection,
    transaction,
    [from],
    {
      maxRetries: 5, // Maximum number of times for the RPC node to retry sending the transaction to the leader 
    },
  );

```
@CriesofCarrots for visibility

### CODE REVIEW

File: web3.js/src/connection.ts [hunk 1]
```diff
@@ -3666,6 +3670,11 @@ export class Connection {
     const preflightCommitment =
       (options && options.preflightCommitment) || this.commitment;
 
+    const maxRetries = options && options.maxRetries;
+
+    if (maxRetries) {
```
Comment (jstarry) [inline web3.js/src/connection.ts:3675]: Could `maxRetries` be zero? That would evaluate to false, which I think would be unexpected. Can you take a look and a test for this?

Comment (RTISCOOL) [reply web3.js/src/connection.ts]: Very true, silly mistake
- fixed below

Comment (t-nelson) [reply web3.js/src/connection.ts]: I think the option is still not applied when the value is zero (retries disabled)?

**Comment by t-nelson** [comment]
Cool!  Once this makes it in, would you mind similarly adding the [`encoding`](https://docs.solana.com/developing/clients/jsonrpc-api#parameters-48) option?

**Pull request closed without merging**